db/compaction_job.cc

```diff
@@ -1498,7 +1498,7 @@ Status CompactionJob::OpenCompactionOutputFile(
       *cfd->ioptions(), *(sub_compact->compaction->mutable_cf_options()),
       cfd->internal_comparator(), cfd->int_tbl_prop_collector_factories(),
       cfd->GetID(), cfd->GetName(), sub_compact->outfile.get(),
-      sub_compact->compaction->output_compression(),
+      sub_compact->compaction->output_compression(), 0,
```

Comment:
Why set sample rate to 0 here?

Reply:
This is the compaction path? I assume the compressibility of data is not going to change in the compaction path. Esp. since the data is sampled, there is no bias for what is retained and what is overwritten. I think the compression goodness we'll report can inform  the decision to compress during compaction too.

Reply:
Discussed offline. Will address this in a separate PR.

Reply:
Add `/*sample_for_compression*/` after `0`.